clang-tools-extra/clangd/refactor/tweaks/ScopifyEnum.cpp

```diff
@@ -125,25 +123,42 @@ llvm::Error ScopifyEnum::addClassKeywordToDeclarations() {
 }
 
 llvm::Error ScopifyEnum::scopifyEnumValues() {
-  std::string PrefixToInsert(D->getName());
-  PrefixToInsert += "::";
+  StringRef EnumName(D->getName());
+  bool StripPrefix = true;
+  for (auto E : D->enumerators()) {
+    if (!E->getName().starts_with(EnumName)) {
+      StripPrefix = false;
+      break;
+    }
+  }
   for (auto E : D->enumerators()) {
```

Comment:
This is not in your changed lines, but please change this `auto` to `const EnumConstantDecl *` as well.